### PR TITLE
Eks roles

### PR DESCRIPTION
**Pull request by alessio-creo**

### CODE REVIEW

File: src/main/61-eks-role.tf [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  role_name = "registryReaderRole"
+
+  attach_vpc_cni_policy = true
+  vpc_cni_enable_ipv4   = true
+
+  oidc_providers = {
+    main = {
+      provider_arn               = module.eks.oidc_provider_arn
+      namespace_service_accounts = ["dev-registry-reader:registry-reader"]
```
[the remark this answers is not in the document]
Comment (alessio-creo) [reply src/main/61-eks-role.tf:12]: I don't think so. Maybe I can use default namespace as placeholder in this phase?

File: src/main/11-iam.tf [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  statement {
+    sid    = "listQueues"
+    effect = "Allow"
+    actions = [
+      "sqs:ListQueues"
+    ]
+    resources = [
+      "*"
+    ]
+  }
```
[the remark this answers is not in the document]
Comment (alessio-creo) [reply src/main/11-iam.tf:42]: We can try. Sometimes ListPrivilges are needed by the sdk

File: src/main/11-iam.tf [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  statement {
+    sid    = "readWriteOnProbingQueue"
+    effect = "Allow"
+    actions = [
+      "s3:GetObject",
+      "s3:ListBucketVersions",
+      "s3:ListBucket",
+      "s3:GetObjectVersion"
```
[the remark this answers is not in the document]
Comment (alessio-creo) [reply src/main/11-iam.tf:25]: Do you mean the versions?

File: src/main/61-eks-role.tf [hunk 4]
```diff
@@ -0,0 +1,22 @@
+module "registry_reader_role" {
+  source = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+
+  role_name = "registryReaderRole"
+
+  attach_vpc_cni_policy = true
+  vpc_cni_enable_ipv4   = true
```
[the remark this answers is not in the document]
Comment (alessio-creo) [reply src/main/61-eks-role.tf:7]: I totally agree

File: src/main/11-iam.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (alessio-creo) [reply src/main/11-iam.tf:25]: understood. I'm going to fix

File: src/main/61-eks-role.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (micdes-pagopa) [reply src/main/61-eks-role.tf:12]: I created the `dev` namespace in the dev cluster, you can use that (the namespace will be the env name)

File: src/main/61-eks-role.tf [hunk 5]
```diff
@@ -0,0 +1,19 @@
+module "registry_reader_role" {
+  source = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+
+  role_name = "${var.app_name}-registry-reader-role-${var.env}"
```
Comment (micdes-pagopa) [inline src/main/61-eks-role.tf:4]:
```suggestion
  role_name = "${var.app_name}-registry-reader-${var.env}"
```

File: src/main/99-outputs.tf [hunk 6]
```diff
@@ -1,4 +1,9 @@
 output "aurora_master_password_secret" {
   value       = aws_secretsmanager_secret_version.database_aurora_master_password.version_id
   description = "Secret ID for Aurora DB master password"
+}
+
+output "bucket_reader_role_arn" {
+  value       = module.registry_reader_role.iam_role_arn
+  description = "Role arn of the probing bucket reader podRole"
```
Comment (micdes-pagopa) [inline src/main/99-outputs.tf:8]:
```suggestion
  description = "ARN of the probing bucket reader pod role"
```

File: src/main/61-eks-role.tf [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+  oidc_providers = {
+    main = {
+      provider_arn               = module.eks.oidc_provider_arn
+      namespace_service_accounts = ["${var.env}:registry-reader"]
+    }
+  }
+
+  role_path        = "/infra/eks/pods/"
```
Comment (micdes-pagopa) [inline src/main/61-eks-role.tf:13]:
```suggestion
  role_path        = "/application/eks/pods/"
```

File: src/main/11-iam.tf [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    ]
+
+    resources = [
+      var.interop_probing_bucket_arn,
+      "${var.interop_probing_bucket_arn}/*",
+    ]
+  }
+
+  statement {
+    sid    = "ReadWriteOnProbingQueue"
```
Comment (micdes-pagopa) [inline src/main/11-iam.tf:19]:
```suggestion
    sid    = "WriteOnRegistryQueue"
```

File: src/main/11-iam.tf [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+      module.sqs_registry_queue.queue_arn
+    ]
+  }
+
+
+}
+
+resource "aws_iam_policy" "registry_reader_policy" {
+  name   = "${var.app_name}-be-registry-reader-${var.env}"
+  path   = "/infra/eks/pods/"
```
Comment (micdes-pagopa) [inline src/main/11-iam.tf:35]:
```suggestion
  path   = "/application/eks/pods/"
```

**Review by micdes-pagopa — CHANGES_REQUESTED** [review]

**Review by micdes-pagopa — APPROVED** [review]

**Pull request merged**